pkg/internal/apis/certmanager/v1/defaults.go

```diff
@@ -17,9 +17,34 @@ limitations under the License.
 package v1
 
 import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/runtime"
+
+	cmapi "github.com/jetstack/cert-manager/pkg/apis/certmanager/v1"
 )
 
 func addDefaultingFuncs(scheme *runtime.Scheme) error {
 	return RegisterDefaults(scheme)
 }
+
+func SetDefaults_CertificateSpec(o *cmapi.CertificateSpec) {
+	if o == nil {
+		return
+	}
+	setDefaultRenewBefore(o)
+}
+
+func setDefaultRenewBefore(o *cmapi.CertificateSpec) {
+	if o.RenewBefore != nil {
+		return
+	}
+	if o.Duration == nil {
+		o.RenewBefore = &metav1.Duration{Duration: cmapi.DefaultRenewBefore}
+		return
+	}
+	renewBefore := o.Duration.Duration / 3
```

Comment:
so currently i you set duration to be < 30 days you get a validation error, does this change make that work?